manifests/service.pp

```diff
@@ -9,4 +9,37 @@
     enable   => $filebeat::service_enable,
     provider => $filebeat::service_provider,
   }
+
+  $major_version                  = $filebeat::major_version
+  $systemd_beat_log_opts_override = $filebeat::systemd_beat_log_opts_override
+
+  #make sure puppet client version 6.1+ with filebeat version 7+, running on systemd
+  if ( versioncmp( $major_version, '7'   ) >= 0 and
+      versioncmp( $::clientversion, '6.1' ) >= 0 and
+      $::service_provider == 'systemd' ) {
+
+    unless $systemd_beat_log_opts_override == undef {
+      $ensure_overide = 'present'
+    } else {
+      $ensure_overide = 'absent'
+    }
+
+    ensure_resource('file',
+      $filebeat::systemd_drop_in_dir,
+      {
+        ensure => 'directory',
+      }
+    )
+
+    ensure_resource('file',
```

Comment:
ok that makes sense. I was more worried that some one else would have managed the file resource.. if they had this would not have thrown an error.. just ignored it..

but yes the user should be made aware that this module uses that resource. :+1: